### PR TITLE
PHPStan: Phase 1

### PR DESCRIPTION
**Pull request by srtfisher**
Phase one of the PHPStan rollout. Adding PHPStan to a number of subtrees.

- Converted many references of `Mantle\Container\Container` or the container contract to `Mantle\Contracts\Application`.
- Condensed the exception handler and removed the duplicate for the console.

### CODE REVIEW

**Review by dustinpaluch — APPROVED** [review]
Looks good! I did see a few imports that don't appear to be necessary anymore, like 
https://github.com/alleyinteractive/mantle-framework/pull/382/files#diff-68c5bb5634ab99d8f371f6ca33b6c2ee9bbe888748a7cda4e7296607224d9815R15

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		# - src/mantle/testkit
+		# - src/mantle/view
+		- mantle.php
+
+	ignoreErrors:
+		- "#Unsafe usage of new static#"
+		- "#PHPDoc tag @param references unknown parameter#"
+
+	scanFiles:
+		- %rootDir%/../../php-stubs/wordpress-stubs/wordpress-stubs.php
```
Comment (szepeviktor) [inline phpstan.neon:44]:
```suggestion
		- %rootDir%/../../php-stubs/wordpress-stubs/wordpress-stubs.php
```

This is included in the extension.

File: src/mantle/application/class-application.php [hunk 2]
```diff
@@ -345,18 +345,18 @@ protected function register_base_bindings() {
 		static::set_instance( $this );
 
 		$this->instance( 'app', $this );
-		$this->instance( Container\Container::class, $this );
+		$this->instance( Container::class, $this );
```
Comment (szepeviktor) [inline src/mantle/application/class-application.php:348]: This is a very nice change ❤️

I could land you my anti-backslash gun 🔫

**Comment by szepeviktor** [comment]
You already benefited from PHPStan at this early stage 🛫
I think it will keep the code base fresh.

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		- %rootDir%/../../php-stubs/wp-cli-stubs/wp-cli-stubs.php
+		- %rootDir%/../../php-stubs/wp-cli-stubs/wp-cli-commands-stubs.php
+		- %rootDir%/../../php-stubs/wp-cli-stubs/wp-cli-i18n-stubs.php
```
Comment (szepeviktor) [inline phpstan.neon:47]: Try running without these two.
Usually these are unused.

File: src/mantle/framework/class-alias-loader.php [hunk 4]
```diff
@@ -72,6 +72,8 @@ public function load( $alias ) {
 		if ( isset( $this->aliases[ $alias ] ) ) {
 			return class_alias( $this->aliases[ $alias ], $alias );
 		}
+
+		return null;
```
Comment (szepeviktor) [inline src/mantle/framework/class-alias-loader.php:76]: The callback's signature is ` callback(string $class): void`
https://www.php.net/manual/en/function.spl-autoload-register.php

`spl_autoload_register` itself returns a boolean.

Comment (szepeviktor) [reply src/mantle/framework/class-alias-loader.php:76]: Then you can remove `@phpstan-ignore-line` 💝

File: src/mantle/framework/console/class-hook-usage-command.php [hunk 5]
```diff
@@ -294,7 +294,7 @@ protected function get_cache_file_for_path( string $path ): string {
 	 */
 	protected function set_paths() {
 		if ( ! $this->option( 'search-path' ) ) {
-			$paths = collect( WP_CONTENT_DIR );
+			$paths = collect( defined( 'WP_CONTENT_DIR' ) ? WP_CONTENT_DIR : getcwd() );
```
Comment (szepeviktor) [inline src/mantle/framework/console/class-hook-usage-command.php:297]: This is not included in the extension as it is not for usage in plugins.

Just create a file - e.g. `tests/phpstan/bootstrap.php`
```php
define( 'WP_CONTENT_DIR', '/' );
```
and add it to `bootStrapFile:` in the configuration.

File: src/mantle/framework/exceptions/class-handler.php [hunk 6]
```diff
@@ -163,11 +166,11 @@ protected function context() {
 	}
 
 	/**
-	 * Render an exception into an HTTP response.
+	 * Render an exception into an HTTP response for the web.
 	 *
 	 * @param Request   $request Request object.
 	 * @param Throwable $e Exception thrown.
-	 * @return \Symfony\Component\HttpFoundation\Response
+	 * @return Response|mixed
```
Comment (szepeviktor) [inline src/mantle/framework/exceptions/class-handler.php:173]: There is no `mixed|...`, mixed includes everything.

File: src/mantle/framework/exceptions/class-handler.php [hunk 7]
```diff
@@ -341,7 +357,7 @@ function ( $trace ) {
 	 * Determine if the given exception is an HTTP exception.
 	 *
 	 * @param Throwable $e Exception thrown.
-	 * @return bool
+	 * @return ($e is HttpException ? true : false)
```
Comment (szepeviktor) [inline src/mantle/framework/exceptions/class-handler.php:360]: You can add this conditional thingy for `is_empty`-s too.

File: src/mantle/framework/manifest/class-model-manifest.php [hunk 8]
```diff
@@ -24,7 +24,7 @@ class Model_Manifest {
 	/**
 	 * Manifest from the disk.
 	 *
-	 * @var array
+	 * @var array|null
```
Comment (szepeviktor) [inline src/mantle/framework/manifest/class-model-manifest.php:27]: The base value for an array is `[]`.

[_source_](https://github.com/szepeviktor/very-clean-code/blob/master/README.md#-base-value-of-an-array)

Comment (szepeviktor) [reply src/mantle/framework/manifest/class-model-manifest.php:27]: 🎈 Fun time!

2 elements `['one', 'two']`
1 element `['one']`
base value `[]`
-1 elements `null` 😲

File: src/mantle/http-client/class-response.php [hunk 9]
```diff
@@ -353,9 +351,9 @@ public function cookies(): array {
 	 * Retrieve a specific cookie by name.
 	 *
 	 * @param string $name Cookie name.
-	 * @return WP_HTTP_Cookie
+	 * @return WP_Http_Cookie
```
Comment (szepeviktor) [inline src/mantle/http-client/class-response.php:354]:
```suggestion
	 * @return WP_Http_Cookie|null
```

File: src/mantle/framework/console/class-hook-usage-command.php (hunk 5 above)
Comment (szepeviktor) [reply src/mantle/framework/console/class-hook-usage-command.php:297]: ... I know, there is no `contrent_dir()` in core

**Comment by szepeviktor** [comment]
@srtfisher What a great improvement!
On PHPStan level 4 you get my seal of approval: _This is a non-broken framework_
Now we are on level 5 🤯

**Comment by srtfisher** [comment]
@szepeviktor I plan on getting all of it all PHPStan and then bumping up the levels. Hopefully it'll get better with each level

**Pull request merged**